File: src/backend/appointment-booking/src/main/java/com/appointment/booking/model/PageLink.java
package com.appointment.booking.model;


import com.appointment.booking.enums.SortOrder;
import java.util.List;
import java.util.Map;
import lombok.Data;

@Data
public class PageLink {

    private int page;
    private int pageSize;
    private String sortProperty;
    private SortOrder sortOrder;
    private String timeZone;
    private Map<String, List<FilterModel>> filters;
    private GlobalFilter globalFilter;

}

File: src/frontend/back-office/src/app/core/models/page-link.ts
import {SortOrder} from "../enum/sort-order.enum";
import {FilterModel} from "./filter-model";
import {GlobalFilter} from "./global-filter";

export interface PageLink {
  page: number;
  pageSize: number;
  sortProperty?: string;
  sortOrder?: SortOrder;
  timeZone?: string;
  filters?: { [key: string]: FilterModel[] };
  globalFilter?: GlobalFilter;
}
